plugins/modules/identity/keycloak/keycloak_client_rolemapping.py

```diff
@@ -0,0 +1,340 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: keycloak_client_rolemapping
+
+short_description: Allows administration of Keycloak client_scopes via Keycloak API
+
+description:
+    - This module allows you to add, remove or modify Keycloak client_rolemapping via the Keycloak REST API.
+      It requires access to the REST API via OpenID Connect; the user connecting and the client being
+      used must have the requisite access rights. In a default Keycloak installation, admin-cli
+      and an admin user would work, as would a separate client definition with the scope tailored
+      to your needs and a user having the expected roles.
+
+    - The names of module options are snake_cased versions of the camelCase ones found in the
+      Keycloak API and its documentation at U(https://www.keycloak.org/docs-api/8.0/rest-api/index.html).
+
+    - Attributes are multi-valued in the Keycloak API. All attributes are lists of individual values and will
+      be returned that way by this module. You may pass single values for attributes when calling the module,
+      and this will be translated into a list suitable for the API.
+
+    - When updating a client_rolemapping, where possible provide the role ID to the module. This removes a lookup
+      to the API to translate the name into the role ID.
+
+
+options:
+    state:
+        description:
+            - State of the client_scope.
+            - On C(present), the client_rolemapping will be created if it does not yet exist, or updated with the parameters you provide.
+            - On C(absent), the client_rolemapping will be removed if it exists.
+        default: 'present'
+        type: str
+        choices:
+            - present
+            - absent
+
+    realm:
+        type: str
+        description:
+            - They Keycloak realm under which this role_representation resides.
+        default: 'master'
+
+    group_name:
+        type: str
+        description:
+            - Name of the group to be mapped.
+            - This parameter is required (can be replaced by gid for less API call).
+
+    gid:
+        type: str
+        description:
+            - Id of the group to be mapped
+            - This parameter is not required for updating or deleting the rolemapping but
+              providing it will reduce the number of API calls required.
+
+    client_id:
+        type: str
+        description:
+            - Name of the client to be mapped (different than cid).
+            - This parameter is required (can be replaced by cid for less API call).
```

Comment:
```suggestion
            - This parameter is required (can be replaced by I(cid) for less API calls).
```
Also: if it is required, why is there no `required: true`?

Reply:
@Gaetan2907 please don't forget this one :)